recipes/llvm-core/all/patches/19x/0000-cmake-dependencies.patch

```diff
@@ -0,0 +1,93 @@
+diff --git a/cmake/config-ix.cmake b/cmake/config-ix.cmake
+index 0aae13e30f2a..c87f22d28dc6 100644
+--- a/cmake/config-ix.cmake
++++ b/cmake/config-ix.cmake
+@@ -137,7 +137,7 @@ if(LLVM_ENABLE_ZLIB)
+     # library on a 64-bit system which would result in a link-time failure.
+     cmake_push_check_state()
+     list(APPEND CMAKE_REQUIRED_INCLUDES ${ZLIB_INCLUDE_DIRS})
+-    list(APPEND CMAKE_REQUIRED_LIBRARIES ${ZLIB_LIBRARY})
++    list(APPEND CMAKE_REQUIRED_LIBRARIES ${ZLIB_LIBRARIES})
+     check_symbol_exists(compress2 zlib.h HAVE_ZLIB)
+     cmake_pop_check_state()
+     if(LLVM_ENABLE_ZLIB STREQUAL FORCE_ON AND NOT HAVE_ZLIB)
+@@ -232,11 +232,11 @@ if(NOT LLVM_USE_SANITIZER MATCHES "Memory.*")
+     # Skip libedit if using ASan as it contains memory leaks.
+     if (LLVM_ENABLE_LIBEDIT AND NOT LLVM_USE_SANITIZER MATCHES ".*Address.*")
+       if(LLVM_ENABLE_LIBEDIT STREQUAL FORCE_ON)
+-        find_package(LibEdit REQUIRED)
++        find_package(editline REQUIRED)
```

Comment:
Patching editline can be avoided with CMakeDeps directly: 

```
deps = CMakeDeps(self)
deps.set_property("editline", "cmake_file_name", "LibEdit")
deps.set_property("editline", "cmake_target_name", "LibEdit::LibEdit")
deps.set_property("editline", "cmake_additional_variables_prefixes", "LibEdit")
```

Reply:
Done